pytorch_lightning/loops/training_epoch_loop.py

```diff
@@ -151,7 +156,7 @@ def on_advance_end(self):
         should_check_val = self.should_check_val_fx(self.iteration_count, self.is_last_batch)
         if should_check_val:
             self.trainer.validating = True
-            self.trainer._run_evaluation()
+            self.trainer.fit_loop.validation_loop.run()
```

Comment:
careful, before, run_evaluation would 1) reset the val dataloader 2) torch.no_grad.

Reply:
Done !